addons/mail/static/src/js/document_viewer.js

```diff
@@ -41,7 +41,12 @@ var DocumentViewer = Widget.extend({
         this._super.apply(this, arguments);
         this.attachment = _.filter(attachments, function (attachment) {
             var type = attachment.mimetype.split('/').shift();
-            if (type === 'image' || type === 'video') {
+
+            if (type === 'application') {
+                type = attachment.mimetype.split('/')[1];
+            }
+
+            if (_.contains(['image', 'video', 'pdf'], type)) {
```

Comment:
At this point it would probably be smarter to pattern-match the mimetype properly instead of devolving into whatever this is, e.g. `/image\/|video\/|application\/pdf/.exec(attachment.mimetype)`